README.md

```diff
@@ -1,144 +1,1607 @@
+
+```go
+config := hazelcast.NewConfig()
+networkConfig := config.NetworkConfig()
+networkConfig.SetRedoOperation(true)
+```
+
+Its default value is `false` (disabled).
+
+## 5.4. Setting Connection Timeout
+
+Connection timeout is the timeout value in milliseconds for members to accept client connection requests.
+If server does not respond within the timeout, the client will retry to connect as many as `ClientNetworkConfig.connectionAttemptLimit` times.
+
+The following are the example configurations.
+
+```go
+config := hazelcast.NewConfig()
+networkConfig := config.NetworkConfig()
+networkConfig.SetConnectionTimeout(6 * time.Second)
+```
+
+Its default value is `5000` milliseconds.
+
+## 5.5. Setting Connection Attempt Limit
+
+While the client is trying to connect initially to one of the members in the `ClientNetworkConfig.addresses`, that member might not be available at that moment. Instead of giving up, throwing an error and stopping the client, the client will retry as many as `ClientNetworkConfig.connectionAttemptLimit` times. This is also the case when the previously established connection between the client and that member goes down.
+
+The following are example configurations.
+
+```go
+config := hazelcast.NewConfig()
+networkConfig := config.NetworkConfig()
+networkConfig.SetConnectionAttemptLimit(5)
+```
+
+Its default value is `2`.
+
+## 5.6. Setting Connection Attempt Period
+
+Connection timeout period is the duration in milliseconds between the connection attempts 
+
+The following are example configurations.
+
+```go
+config := hazelcast.NewConfig()
+networkConfig := config.NetworkConfig()
+networkConfig.SetConnectionAttemptPeriod(5 * time.Second)
+```
+
+Its default value is `3000` milliseconds.
+
+## 5.7. Enabling Client TLS/SSL
+
+You can use TLS/SSL to secure the connection between the clients and members. If you want TLS/SSL enabled
+for the client-cluster connection, you should set an SSL configuration. Please see [TLS/SSL section](#1-tlsssl).
+
+As explained in the [TLS/SSL section](#1-tlsssl), Hazelcast members have key stores used to identify themselves (to other members) and Hazelcast Go clients have certificate authorities used to define which members they can trust. Hazelcast has the mutual authentication feature which allows the Go clients also to have their private keys and public certificates and members to have their certificate authorities so that the members can know which clients they can trust. Please see the [Mutual Authentication section](#13-mutual-authentication).
+
+## 5.8. Enabling Hazelcast Cloud Discovery
+
+The purpose of Hazelcast Cloud Discovery is to provide clients to use IP addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, specify a token for the `discoveryToken` field and set the `enabled` field to `true`.
```

Comment:
I think we should talk about Hazelcast Cloud here, or maybe put some links and the first sentence should be changed to something like : `The purpose of Hazelcast Cloud Discovery is to discover cloud IP addresses provided the hazelcast orchestrator`